### PR TITLE
Migrate from Maven to Gradle

### PR DESCRIPTION
**Pull request by danielpassos**
Related issues: 
* [AGDROID-615](https://issues.jboss.org/browse/AGDROID-615) - Migrate aerogear-android-sync from Maven to Gradle
* [AGDROID-626](https://issues.jboss.org/browse/AGDROID-626) - Enable circle-ci for aerogear-android-sync

### CODE REVIEW

File: library/build.gradle [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    }
+}
+
+dependencies {
+    compile 'org.jboss.aerogear:sync-json-patch-core:1.0.0-alpha.1'
+    compile 'org.jboss.aerogear:sync-json-patch-client:1.0.0-alpha.1'
+    compile 'org.jboss.aerogear:sync-client-netty:1.0.0-alpha.1'
+}
+
+apply from: '../gradle-mvn-push.gradle'
```
Comment (wojta) [inline library/build.gradle]: I usually have all 'apply' stuff on top for clarit.

Comment (danielpassos) [inline library/build.gradle]: Me too, but this is a recommendation from the author, same of the Firebase

**Comment by secondsun** [comment]
When I build I get a 

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':library:transformResourcesWithMergeJavaResForDebugAndroidTest'.
> com.android.build.api.transform.TransformException: com.android.builder.packaging.DuplicateFileException: Duplicate files copied in APK META-INF/LICENSE
        File1: /home/summers/.gradle/caches/modules-2/files-2.1/com.github.fge/json-patch/1.9/a4c3c97a0f5965dec15795acf40d3fbc897af4b/json-patch-1.9.jar
        File2: /home/summers/.gradle/caches/modules-2/files-2.1/com.fasterxml.jackson.core/jackson-annotations/2.3.0/f5e853a20b60758922453d56f9ae1e64af5cb3da/jackson-annotations-2.3.0.jar
        File3: /home/summers/.gradle/caches/modules-2/files-2.1/com.fasterxml.jackson.core/jackson-core/2.3.0/5e19d8381e01aa64c9dd47ff453e39abc441775c/jackson-core-2.3.0.jar
        File4: /home/summers/.gradle/caches/modules-2/files-2.1/com.github.fge/jackson-coreutils/1.6/9e6af56eb7cc2a65700b289abc7ee2bd170fd231/jackson-coreutils-1.6.jar
        File5: /home/summers/.gradle/caches/modules-2/files-2.1/com.fasterxml.jackson.core/jackson-databind/2.3.0/76eb119e9f7769c5b124afbfa17ed0c63cab4920/jackson-databind-2.3.0.jar
        File6: /home/summers/.gradle/caches/modules-2/files-2.1/com.github.fge/msg-simple/1.1/f261263e13dd4cfa93cc6b83f1f58f619097a2c4/msg-simple-1.1.jar
        File7: /home/summers/.gradle/caches/modules-2/files-2.1/com.github.fge/btf/1.2/9e66651022eb86301b348d57e6f59459effc343b/btf-1.2.jar
```
passos, it looks like we need to exclude some resources during the build.

**Comment by danielpassos** [comment]
@secondsun Weird

```
[passos]: ~/Development/Projects/AeroGear/aerogear-android-offline [git:gradle]
→ gradle clean build
Starting a Gradle Daemon (subsequent builds will be faster)
:clean
:library:clean
:library:preBuild UP-TO-DATE
:library:preDebugBuild UP-TO-DATE
:library:checkDebugManifest
:library:preDebugAndroidTestBuild UP-TO-DATE
:library:preDebugUnitTestBuild UP-TO-DATE
:library:preReleaseBuild UP-TO-DATE
:library:preReleaseUnitTestBuild UP-TO-DATE
:library:prepareComAndroidSupportSupportV42340Library
:library:prepareOrgJbossAerogearAerogearAndroidCore300Library
:library:prepareOrgJbossAerogearAerogearAndroidPipe310Library
:library:prepareDebugDependencies
:library:compileDebugAidl
:library:compileDebugNdk NO-SOURCE
:library:compileLint
:library:copyDebugLint NO-SOURCE
:library:compileDebugRenderscript
:library:generateDebugBuildConfig
:library:generateDebugResValues
:library:generateDebugResources
:library:mergeDebugResources
:library:processDebugManifest
:library:processDebugResources
:library:generateDebugSources
:library:incrementalDebugJavaCompilationSafeguard
:library:javaPreCompileDebug
:library:compileDebugJavaWithJavac
:library:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: /Users/passos/Development/Projects/AeroGear/aerogear-android-offline/library/src/main/java/org/jboss/aerogear/android/offline/StorageConfiguration.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:library:extractDebugAnnotations
:library:mergeDebugShaders
:library:compileDebugShaders
:library:generateDebugAssets
:library:mergeDebugAssets
:library:mergeDebugProguardFiles
:library:packageDebugRenderscript NO-SOURCE
:library:packageDebugResources
:library:processDebugJavaRes NO-SOURCE
:library:transformResourcesWithMergeJavaResForDebug
:library:transformClassesAndResourcesWithSyncLibJarsForDebug
:library:mergeDebugJniLibFolders
:library:transformNativeLibsWithMergeJniLibsForDebug
:library:transformNativeLibsWithStripDebugSymbolForDebug
:library:transformNativeLibsWithSyncJniLibsForDebug
:library:bundleDebug
:library:compileDebugSources
:library:assembleDebug
:library:checkReleaseManifest
:library:prepareReleaseDependencies
:library:compileReleaseAidl
:library:compileReleaseNdk NO-SOURCE
:library:copyReleaseLint NO-SOURCE
:library:compileReleaseRenderscript
:library:generateReleaseBuildConfig
:library:generateReleaseResValues
:library:generateReleaseResources
:library:mergeReleaseResources
:library:processReleaseManifest
:library:processReleaseResources
:library:generateReleaseSources
:library:incrementalReleaseJavaCompilationSafeguard
:library:javaPreCompileRelease
:library:compileReleaseJavaWithJavac
:library:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: /Users/passos/Development/Projects/AeroGear/aerogear-android-offline/library/src/main/java/org/jboss/aerogear/android/offline/StorageConfiguration.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:library:extractReleaseAnnotations
:library:mergeReleaseShaders
:library:compileReleaseShaders
:library:generateReleaseAssets
:library:mergeReleaseAssets
:library:mergeReleaseProguardFiles
:library:packageReleaseRenderscript NO-SOURCE
:library:packageReleaseResources
:library:processReleaseJavaRes NO-SOURCE
:library:transformResourcesWithMergeJavaResForRelease
:library:transformClassesAndResourcesWithSyncLibJarsForRelease
:library:mergeReleaseJniLibFolders
:library:transformNativeLibsWithMergeJniLibsForRelease
:library:transformNativeLibsWithStripDebugSymbolForRelease
:library:transformNativeLibsWithSyncJniLibsForRelease
:library:bundleRelease
:library:compileReleaseSources
:library:assembleRelease
:library:generateDebugJavadoc
:library:jarDebugJavadoc
:library:jarDebugSources
:library:generateReleaseJavadoc
:library:jarReleaseJavadoc
:library:jarReleaseSources
:library:signArchives
:library:assemble
:library:lint
Ran lint on variant release: 2 issues found
Ran lint on variant debug: 2 issues found
Wrote HTML report to file:///Users/passos/Development/Projects/AeroGear/aerogear-android-offline/library/build/reports/lint-results.html
Wrote XML report to file:///Users/passos/Development/Projects/AeroGear/aerogear-android-offline/library/build/reports/lint-results.xml
:library:incrementalDebugUnitTestJavaCompilationSafeguard NO-SOURCE
:library:javaPreCompileDebugUnitTest
:library:prepareDebugUnitTestDependencies
:library:compileDebugUnitTestJavaWithJavac NO-SOURCE
:library:processDebugUnitTestJavaRes NO-SOURCE
:library:compileDebugUnitTestSources UP-TO-DATE
:library:mockableAndroidJar
:library:assembleDebugUnitTest
:library:testDebugUnitTest NO-SOURCE
:library:incrementalReleaseUnitTestJavaCompilationSafeguard NO-SOURCE
:library:javaPreCompileReleaseUnitTest
:library:prepareReleaseUnitTestDependencies
:library:compileReleaseUnitTestJavaWithJavac NO-SOURCE
:library:processReleaseUnitTestJavaRes NO-SOURCE
:library:compileReleaseUnitTestSources UP-TO-DATE
:library:assembleReleaseUnitTest
:library:testReleaseUnitTest NO-SOURCE
:library:test UP-TO-DATE
:library:check
:library:build

BUILD SUCCESSFUL

Total time: 22.569 secs
```

**Comment by danielpassos** [comment]
@secondsun Weird 

```[passos]: ~/Development/Projects/AeroGear/aerogear-android-sync [git:gradle]
→ gradle clean build
:clean
:library:clean
:library:preBuild UP-TO-DATE
:library:preDebugBuild UP-TO-DATE
:library:checkDebugManifest
:library:prepareDebugDependencies
:library:compileDebugAidl
:library:compileDebugNdk NO-SOURCE
:library:compileLint
:library:copyDebugLint NO-SOURCE
:library:mergeDebugShaders
:library:compileDebugShaders
:library:generateDebugAssets
:library:mergeDebugAssets
:library:mergeDebugProguardFiles
:library:packageDebugRenderscript NO-SOURCE
:library:compileDebugRenderscript UP-TO-DATE
:library:generateDebugResValues UP-TO-DATE
:library:generateDebugResources UP-TO-DATE
:library:packageDebugResources
:library:processDebugManifest
:library:generateDebugBuildConfig
:library:processDebugResources
:library:generateDebugSources
:library:incrementalDebugJavaCompilationSafeguard
:library:javaPreCompileDebug
:library:compileDebugJavaWithJavac
:library:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
:library:processDebugJavaRes NO-SOURCE
:library:transformResourcesWithMergeJavaResForDebug
:library:transformClassesAndResourcesWithSyncLibJarsForDebug
:library:mergeDebugJniLibFolders
:library:transformNativeLibsWithMergeJniLibsForDebug
:library:transformNativeLibsWithStripDebugSymbolForDebug
:library:transformNativeLibsWithSyncJniLibsForDebug
:library:bundleDebug
:library:compileDebugSources
:library:assembleDebug
:library:preReleaseBuild UP-TO-DATE
:library:checkReleaseManifest
:library:prepareReleaseDependencies
:library:compileReleaseAidl
:library:compileReleaseNdk NO-SOURCE
:library:copyReleaseLint NO-SOURCE
:library:mergeReleaseShaders
:library:compileReleaseShaders
:library:generateReleaseAssets
:library:mergeReleaseAssets
:library:mergeReleaseProguardFiles
:library:packageReleaseRenderscript NO-SOURCE
:library:compileReleaseRenderscript
:library:generateReleaseResValues
:library:generateReleaseResources
:library:packageReleaseResources
:library:processReleaseManifest
:library:generateReleaseBuildConfig
:library:processReleaseResources
:library:generateReleaseSources
:library:incrementalReleaseJavaCompilationSafeguard
:library:javaPreCompileRelease
:library:compileReleaseJavaWithJavac
:library:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
:library:processReleaseJavaRes NO-SOURCE
:library:transformResourcesWithMergeJavaResForRelease
:library:transformClassesAndResourcesWithSyncLibJarsForRelease
:library:mergeReleaseJniLibFolders
:library:transformNativeLibsWithMergeJniLibsForRelease
:library:transformNativeLibsWithStripDebugSymbolForRelease
:library:transformNativeLibsWithSyncJniLibsForRelease
:library:bundleRelease
:library:compileReleaseSources
:library:assembleRelease
:library:generateDebugJavadoc
:library:jarDebugJavadoc
:library:jarDebugSources
:library:generateReleaseJavadoc
:library:jarReleaseJavadoc
:library:jarReleaseSources
:library:signArchives
:library:assemble
:library:lint
Ran lint on variant debug: 1 issues found
Ran lint on variant release: 1 issues found
Wrote HTML report to file:///Users/passos/Development/Projects/AeroGear/aerogear-android-sync/library/build/reports/lint-results.html
Wrote XML report to file:///Users/passos/Development/Projects/AeroGear/aerogear-android-sync/library/build/reports/lint-results.xml
:library:incrementalDebugUnitTestJavaCompilationSafeguard NO-SOURCE
:library:javaPreCompileDebugUnitTest
:library:preDebugUnitTestBuild UP-TO-DATE
:library:prepareDebugUnitTestDependencies
:library:compileDebugUnitTestJavaWithJavac NO-SOURCE
:library:processDebugUnitTestJavaRes NO-SOURCE
:library:compileDebugUnitTestSources UP-TO-DATE
:library:mockableAndroidJar
:library:assembleDebugUnitTest
:library:testDebugUnitTest NO-SOURCE
:library:incrementalReleaseUnitTestJavaCompilationSafeguard NO-SOURCE
:library:javaPreCompileReleaseUnitTest
:library:preReleaseUnitTestBuild UP-TO-DATE
:library:prepareReleaseUnitTestDependencies
:library:compileReleaseUnitTestJavaWithJavac NO-SOURCE
:library:processReleaseUnitTestJavaRes NO-SOURCE
:library:compileReleaseUnitTestSources UP-TO-DATE
:library:assembleReleaseUnitTest
:library:testReleaseUnitTest NO-SOURCE
:library:test UP-TO-DATE
:library:check
:library:build

BUILD SUCCESSFUL

Total time: 8.375 secs
```

**Comment by secondsun** [comment]
It happens when running the tests.  But there are no tests so I guess I need to look deeper.

**Pull request merged**